src/lightning_app/utilities/proxies.py

```diff
@@ -420,6 +421,9 @@ def __call__(self):
     def setup(self):
         from lightning_app.utilities.state import AppState
 
+        # Note: Enable to capture all signals and call them sequentially in the order they got registered
+        signal.signal = SignalHandlerCompose(signal.signal)
```

Comment:
What if signal.signal is already a SignalHandlerCompose? You will get `SignalHandlerCompose(SignalHandlerCompose(SignalHandlerCompose(SignalHandlerCompose(....`
since this is a global setting and so you will stack it across test runs, right?